File: DSA/Arrays/carry-forward/leaders-in-array/python/leadersInArray.py
def leadersInArray(self, A):
        leader = [A[-1]]
        max = A[-1]
        for i in range(len(A)-2, -1, -1):
            if A[i] > max:
                leader.append(A[i])
                max = A[i]
        
        return leader

File: DSA/Arrays/carry-forward/leaders-in-array/javascript/leadersInArray.js
export const leadersInArray = (A) => {
    const leaders = [A[A.length - 1]];
    let currMax = A[A.length - 1];
    for (let i = A.length - 2; i >= 0; i--) {
        if (currMax < A[i]) {
            leaders.push(A[i]);
            currMax = A[i];
        }
    }
    return leaders;
};
